language-support/java/codegen/src/main/scala/com/digitalasset/daml/lf/codegen/CodeGenRunner.scala

```diff
@@ -180,68 +259,27 @@ object CodeGenRunner extends StrictLogging {
     }
   }
 
-  private[CodeGenRunner] def generateCode(
-      interfaces: Seq[Interface],
-      conf: Conf,
-      pkgPrefixes: Map[PackageId, String],
-  )(implicit ec: ExecutionContext): Unit = {
-    logger.info(
-      s"Start processing packageIds '${interfaces.map(_.packageId).mkString(", ")}' in directory '${conf.outputDirectory}'"
-    )
-
-    val prefixes = resolvePackagePrefixes(pkgPrefixes, conf.modulePrefixes, interfaces)
-    detectModuleCollisions(prefixes, interfaces)
-
-    // TODO (mp): pre-processing and escaping
-    val preprocessingFuture: Future[InterfaceTrees] =
-      backend.preprocess(interfaces, conf, prefixes)
-
-    val future: Future[Unit] = {
-      for {
-        preprocessedInterfaceTrees <- preprocessingFuture
-        _ <- Future.traverse(preprocessedInterfaceTrees.interfaceTrees)(
-          processInterfaceTree(_, conf, prefixes)
-        )
-
-      } yield ()
-    }
-
-    // TODO (mp): make the timeout configurable
-    val _ = Await.result(future, Duration.create(10L, TimeUnit.MINUTES))
-    logger.info(s"Finish processing packageIds ''${interfaces.map(_.packageId).mkString(", ")}''")
-  }
-
-  // TODO (#584): Make Java Codegen Backend configurable
-  private[codegen] val backend: Backend = JavaBackend
-
-  private[CodeGenRunner] def processInterfaceTree(
```

Comment:
This could have used some context and has been moved to the companion class to this object.